docs/src/your-own-sender.md

```diff
@@ -0,0 +1,197 @@
+---
+short-description: Brief overview of a custom sender and operation.
+...
+
+# Writing your own sender and operation
+
+While libasync provides a veriaty of existing types, one may wish to write their own senders and operations. The most common reason for writing a custom sender and operation is wrapping around an existing callback-based API while avoiding extra costs induced by allocating coroutine frames. The following section goes into detail on how to implement them by implementing a simple wrapper for libuv's `uv_write`.
+
+## End effect
+
+The sender and operation we'll implement here will allow us to do the following:
+```cpp
+auto result = co_await write(my_handle, my_bufs, n_my_bufs);
+if (result < 0)
+	/* report error */
+```
+
+## The implementation
+
+### The sender
+
+As explained in [](sender-receiver.md), the sender is an object that stores the neceessary state to start an operation. Let's start off by looking at the prototype for `uv_write`:
```

Comment:
does an unnamed link work properly?

Reply:
Yes, hotdoc will use the page title instead.